### PR TITLE
FIX: #228 Creatures doing a little "wait step" on reaching wp

### PR DESCRIPTION
**Pull request by dhmann**
**FIX for ISSUE 228**

As stated in issue 228, the little "wait step" happened because the orientation gets set to 0 on reaching a waypoint. Upon examining the issue, I found out orientation was set to 0 at several places where it made no sense.

I have taken a look at several videos on youtube from vanilla to wotlk and also compared our Waypoint and MoveTo functions with other emulators.

While doing so I made the following observations:

-Units don't need to be given an orientation to move. The orientation is given from the direction the unit is facing while moving, so we can remove orientation as an argument for MoveTo and Move.
-Most times orientation was given as 0 anyways, so it was broken from the start...
-There seem to be no cases where a unit needs to look into another direction exactly as he reaches the destination. In the rare cases where we might indeed need to do something like this, we should rather use setOrientation() or setFacing() instead of using MoveTo() to accomplish this.

What I did:

I simply rewrote the MoveTo and Move Functions to no longer use orientantion. Orientation can be set manually by setOrientation() or setFacing() when needed.

How to reproduce the fix:

Make a new Human Character and take a look at the guards patrolling around.

What still needs to be done:

I still don't think Waypoints work 100% as intended.

The wait step bug seems to be fixed but I think I opened a bigger can of worms. The Waypointsystem is messy, lots of dead and commented code. I might take a further look into this.

### CODE REVIEW

File: src/scripts/LuaEngine/UnitFunctions.h [hunk 1]
```diff
@@ -2035,15 +2035,23 @@ class LuaUnit
...
+        }
 
         float x = ptr->GetSpawnX();
         float y = ptr->GetSpawnY();
         float z = ptr->GetSpawnZ();
         float o = ptr->GetSpawnO();
-        if (ptr->IsCreature())
-            ptr->GetAIInterface()->MoveTo(x, y, z, o);
+
+		if (ptr->IsCreature())
```
Comment (Zyres) [inline src/scripts/LuaEngine/UnitFunctions.h]: Tabs

File: src/scripts/LuaEngine/UnitFunctions.h [hunk 2]
```diff
@@ -2035,15 +2035,23 @@ class LuaUnit
 
     static int ReturnToSpawnPoint(lua_State* L, Unit* ptr)
     {
-        if (ptr == nullptr)
+
+		if (ptr == nullptr)
```
Comment (Zyres) [inline src/scripts/LuaEngine/UnitFunctions.h]: Tabs.

File: src/scripts/LuaEngine/UnitFunctions.h (hunk 1 above)
Comment (dhmann) [inline src/scripts/LuaEngine/UnitFunctions.h]: Sure man, will do when I'll get back from university :) I am sorry I had a few problems with Visual Studio, last one I used was 2010 :)

Comment (Zyres) [inline src/scripts/LuaEngine/UnitFunctions.h]: :-) You still miss line 2039

Comment (dhmann) [inline src/scripts/LuaEngine/UnitFunctions.h]: Goddamn, don't commit before your fist coffee... Will recommit when I am home and got my hands on my sourcefiles. :-)

**Comment by dhmann** [comment]
The Wait Step should now be finally fixed. Issue seemed to be we didn't update the MovementSpline when we reached our target.

**Pull request merged**